qiskit_ibm_runtime/utils/noise_learner_result.py

```diff
@@ -14,47 +14,136 @@
 
 from __future__ import annotations
 
-from dataclasses import dataclass
-from typing import Any, Iterator, Sequence
+from typing import Any, Iterator, List, Sequence
+from numpy.typing import NDArray
+import numpy as np
 
 from qiskit.circuit import QuantumCircuit
 from qiskit.quantum_info import PauliList
 
 
-@dataclass(frozen=True)
-class NoiseLearnerDatum:
-    """A container for a single noise learner datum.
+class PauliLindbladError:
+    r"""A Pauli error channel generated by a Pauli Lindblad dissipator.
+
+    This operator represents an N-qubit quantum error channel
+    :math:`E(ρ) = e^{\sum_j r_j D_{P_j}}(ρ)` generated by Pauli Lindblad dissipators
+    :math:`D_P(ρ) = P ρ P - ρ`, where :math:`P_j` are N-qubit :class:`~.Pauli` operators.
+
+    The list of Pauli generator terms are stored as a :class:`~.PauliList` and can be
+    accessed via the :attr:`generators` attribute. The array of dissipator rates
+    :math:`r_j` can be accessed via the :attr:`rates` attribute.
+
+    The equivalent Pauli error channel can be constructed as a composition
+    of single-Pauli channel terms
+
+    .. math::
+
+        E = e^{\sum_j r_j D_{P_j}} = \prod_j e^{r_j D_{P_j}}
+        = prod_j \left( (1 - p_j) S_I + p_j S_{P_j} \right)
+
+    where :math:`p_j = \frac12 - \frac12 e^{-2 r_j}`.
 
     Args:
-        circuit: A circuit whose noise has been learnt.
-        qubits: The labels of the qubits in the given circuit.
-        generators: The list of generators for the Pauli errors.
-        rates: The rates for the given Pauli error generators.
+        generators: A list of the Pauli Lindblad generators for the error channel.
+        rates: A list of the rates for the Pauli-Lindblad ``generators``.
 
     Raises:
-        ValueError: If ``circuit``, ``qubits``, and ``generators`` have mismatching number of
-            qubits.
         ValueError: If ``generators`` and ``rates`` have different lengths.
     """
 
-    circuit: QuantumCircuit
-    qubits: Sequence[int]
-    generators: PauliList
-    rates: Sequence[float]
+    def __init__(self, generators: PauliList, rates: Sequence[float]) -> None:
+        self._generators = generators
+        self._rates = np.asarray(rates, dtype=float)
 
-    def __post_init__(self) -> None:
-        if len({self.circuit.num_qubits, len(self.qubits), self.generators.num_qubits}) != 1:
-            raise ValueError("Mistmatching numbers of qubits.")
-        if len(self.generators) != len(self.rates):
-            msg = f"``generators`` has length {len(self.generators)}, but "
-            msg += f"``rates`` has length {len(self.rates)}."
+        if len(generators) != len(self._rates):
+            msg = f"``generators`` has length {len(generators)} "
+            msg += f"but ``rates`` has length {len(self.rates)}."
             raise ValueError(msg)
```

Comment:
This might catch more edge cases
```suggestion
        if (len(generators),) != self._rates.shape:
            raise ValueError(
                f"``generators`` has length {len(generators)} "
                f"but ``rates`` has shape {self._rates.shape}."
            )
```

Reply:
True